app/src/main/java/de/westnordost/streetcomplete/quests/smoking/SmokingAllowedAnswerForm.kt

```diff
@@ -0,0 +1,25 @@
+package de.westnordost.streetcomplete.quests.smoking
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.quests.AbstractQuestAnswerFragment
+import de.westnordost.streetcomplete.quests.AnswerItem
+import de.westnordost.streetcomplete.quests.smoking.SmokingAllowed.YES
+import de.westnordost.streetcomplete.quests.smoking.SmokingAllowed.OUTSIDE
+import de.westnordost.streetcomplete.quests.smoking.SmokingAllowed.NO
+import de.westnordost.streetcomplete.quests.smoking.SmokingAllowed.ISOLATED
+import de.westnordost.streetcomplete.quests.smoking.SmokingAllowed.SEPARATED
+
+class SmokingAllowedAnswerForm : AbstractQuestAnswerFragment<SmokingAllowed>() {
+
+    override val buttonPanelAnswers = listOf(
+        AnswerItem(R.string.quest_generic_hasFeature_no) { applyAnswer(NO) },
+        AnswerItem(R.string.quest_generic_hasFeature_yes) { applyAnswer(YES) },
```

Comment:
I need to check how it works in practice, but I suspect that with "Is smoking allowed anywhere in (...)" question people may answer "yes" where ISOLATED/SEPARATED would be better answer.

Maybe replace "Yes" by "Everywhere" option? Or display all answers (like playground access and parking access quests)?

Reply:
good idea, I've tried "parking access" type now in e0a4b6d4efca2fb0fae7d0f293172153fec6b00b